### PR TITLE
Added Delete button for individual Client records

### LINKED ISSUE #2393: [UI] Add a delete action in roles, clients, secrets, hooks views
Feature initially proposed in https://github.com/taskcluster/taskcluster/issues/2386#issue-568468798.

**Describe the bug**
In order to delete something from (e.g., https://community-tc.services.mozilla.com/auth/roles), a user needs to click on the role to delete, then click/hover on the triple dot fab button on the lower right then click on "Delete". That's a lot of clicks that we can perhaps avoid by having the list of hooks show a delete icon beside the name.

**To Reproduce**

1. Go to https://community-tc.services.mozilla.com/auth/roles

**Expected behavior**
A delete icon is shown beside each role. Clicking on the delete icon should behave the same way as the one in the speed dial (triple dots). In other words, it should first show a confirmation dialog prior to performing the actual delete.

**Taskcluster version**
v24.3.0

**Additional context**
Once we have a good idea about the implementation, we'll probably need to do the same on the following views:
* https://community-tc.services.mozilla.com/auth/clients
* https://community-tc.services.mozilla.com/secrets
* https://community-tc.services.mozilla.com/hooks

### PR DESCRIPTION
**Pull request by mrsVelena**
Github Bug/Issue: Fixes #2393 for Clients page

### CODE REVIEW

File: changelog/issue-2393.md [hunk 1]
```diff
@@ -2,4 +2,6 @@ audience: developers
 level: minor
```
Comment (djmitche) [inline changelog/issue-2393.md:2]: I missed this on the earlier PR, but let's change this to "patch".  The guidelines for these levels are [here](https://github.com/taskcluster/taskcluster/blob/main/dev-docs/best-practices/changelog.md), In this case, "A patch change is one that fixes a bug or makes a minor change which users can do their work without." -- users can do their work without this feature, just a bit more slowly.

**Review by djmitche — CHANGES_REQUESTED** [review]
One minor request: let's move the delete icon to the rightmost end of the list item, so instead of

image

it looks like

image

**Comment by djmitche** [comment]
Ah!  I didn't see the changes you pushed :(  There's a little circular-arrow button next to the reviews in the GitHub UI that will re-request a review, and hopefully get a response a little sooner.  Sorry!  I'll have a look shortly.

**Review by djmitche — CHANGES_REQUESTED** [review]
This looks great, but seems to have one lingering bug. When I click the delete icon, it displays a dialog that says "This will delete the `github/28673|djmitche/cli-ZN_L9m` client" (which is the clientId I was trying to delete).  When I click the "Delete Client" button, the dialog changes to

image

Any idea what's wrong?

**Comment by mrsVelena** [comment]
I'm sorry for the late answer (I was on vacation and didn't have access to my computer). I actually cannot reach this step, because I don't have account with permissions to delete anything. But I assumed, there could be this problem, because I don't close Dialog onComplete after deleting. I tried to fix this, please check if this helped.

**Comment by djmitche** [comment]
That fix seems perfect!  I noticed, though, that the client is still in the list after the deletion occurs.  Is that something you could fix?

Here's how to create clients:
 * On https://community-tc.services.mozilla.com/profile (after signing in) look for a scope like `auth:create-client:github/<something>`; mine is `auth:create-client:github/28673|djmitche/*`.  That gives you permission to create clients with that prefix (In my case, I can create clients like `github/28673|djmitche/foo` and `github/28673|djmitche/bar`).
 * Create a new client for yourself with a name you have permission for; mine might be `github/28673|djmitche/issue-2393`.  Give it scopes to create and delete clients -- for me that would be `auth:create-client:github/28673|djmitche/*` and auth:delete-client:github/28673|djmitche/*`.  Set the expiration to a few months from now.
 * On your development server, sign in with this new clientId and the accessToken it gave you (leave the certificate blank).  Now you can use your development server to create and delete clients all you want!

Sorry that's so complicated -- signing in is always difficult in development environments, it seems.

**Comment by mrsVelena** [comment]
I fixed list update after client deletion. But I think the same bug would be on Roles page, too. What would be a better way to fix it? Through creating new PR?

**Comment by djmitche** [comment]
Good thinking.  Yes, this PR has been around for a while and is (hopefully) ready to merge, so a new one is a good idea.

**Review by djmitche — APPROVED** [review]
And, indeed, this is great -- thanks for your patience and persistence!

**Comment by djmitche** [comment]
(I merged main in here and fixed merge conflicts -- I'll merge back to main when the tests pass)

**Pull request merged**